src/pages/user/UserInstitution.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, FC } from 'react';
+import Card from '../../components/Card';
+import InstitutionCard from './institution/InstitutionCard';
+import { Button } from '@material-ui/core';
+import { useSelector } from 'react-redux';
+import { RootStore } from './../../redux/reducers/rootReducer';
+import styled from 'styled-components';
+import { useTranslation } from 'react-i18next';
+import InstitutionSelector from './institution/InstitutionSelector';
+import Heading from '../../components/Heading';
+import { Formik, FormikProps, Field } from 'formik';
+import InstitutionSearch from '../publication/references_tab/components/InstitutionSearch';
+import {
+  emptyRecursiveUnit,
+  Subunit,
+  Unit,
+  emptyFormikUnitState,
+  FormikUnitFieldNames,
+  FormikUnit,
+} from '../../types/institution.types';
+
+const StyledButtonContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  margin-top: 1rem;
+`;
+
+const StyledButton = styled(Button)`
+  margin: 0.5rem;
+`;
+
+const StyledInstitutionSearchContainer = styled.div`
+  width: 30rem;
+`;
+
+const UserInstitution: FC = () => {
+  const user = useSelector((state: RootStore) => state.user);
+  const [open, setOpen] = useState(false);
+  const [units, setUnits] = useState<Unit[]>([]);
+  const { t } = useTranslation('profile');
+
+  const handleClickAdd = () => {
+    setOpen(true);
+  };
+
+  const handleCancel = () => {
+    setOpen(false);
+  };
+
+  const handleAddInstitution = ({ name, id, subunits }: Unit) => {
+    // TODO: find which is the lower subunit to be saved in ARP
+    // try {
+    //   // update ARP
+    // } catch (error) {
+    //   // handle error
+    // }
+    const filteredSubunits = subunits.filter((u: Subunit) => u.name !== '');
+    setUnits([...units, { id, name, subunits: filteredSubunits }]);
+    setOpen(false);
+  };
+
+  const onSubmit = async (values: FormikUnit, { resetForm }: any) => {
+    handleAddInstitution({ name: values.name, id: values.id, subunits: values.subunits });
+    resetForm({});
+  };
+
+  return (
+    <Card>
+      <Heading>{t('heading.organizations')}</Heading>
+      {units && units.map((unit: Unit, index: number) => <InstitutionCard key={index} unit={unit} />)}
```

Comment:
kan vel bruke `units?.map()` også her

Reply:
kunne kanskje brukt `unit.id` eller cristinId som key også

Reply:
ja, tenkte å gjøre det når jeg sjekker for duplikater (i en fremtidig PR), men ble ikke gjort nå pga key-klaging hvis man legger til samme to ganger. men det skal fikses 👍